rasa/core/agent.py

```diff
@@ -316,6 +314,38 @@ def __init__(
         self.model_server = model_server
         self.remote_storage = remote_storage
 
+    def _check_missing_policies(self):
+        """Check for domain elements that work only with certain policies."""
+
+        def check_form_policy():
```

Comment:
I would not do a function in a function.
Maybe we could outsource the validations to the policies / policyEnsemble themselves.

maybe we can do a static method in the PolicyEnsemble, e.g. `PolicyEnsemble.validate_domain(ensemble, domain)` where we check it. `agent` is crowded enough. What do you think?

Reply:
I got inspiration from this function - https://github.com/RasaHQ/rasa/blob/fbef25988ee059fc14b52ae9c45132c05f344279/rasa/core/domain.py#L764

Reply:
I agree that `PolicyEnsemble` might be a more appropriate place to put it, though.

Reply:
Mhm, I read about it (https://stackabuse.com/python-nested-functions/, https://realpython.com/inner-functions-what-are-they-good-for/ ) and I don't see the case here. Just makes it harder to test.

Reply:
@wochinge Just to make one final case for keeping in part of `Agent`, I'm thinking that since the `Agent` is what brings together all the different configuration files, the responsibility of checking mutual compatibility should also lie with it.